client/components/admin_dashboard/Map.jsx

```diff
@@ -0,0 +1,77 @@
+"use client";
+import { GoogleMap, LoadScript, Marker } from "@react-google-maps/api";
+import { googleApiKey } from "@/constants";
```

Comment:
You can't be accessing server secrets from a client component can you? Bit concerned our dual-server setup is quite confusing for reasons such as this

Reply:
As long as an env variable is prefixed with `NEXT_PUBLIC_` it can be exposed to the browser, which is how we set it up

Reply:
Oh I didn't know about next public. But yeah I think it's still an issue to expose the API key

Reply:
You pretty much have to expose the API key when using Google Maps, since the map itself has to be rendered in the browser. It's still good practice to keep the key hidden in your codebase (in fact Google will alert you if you upload the key to GitHub (ask me how I know!)), which is why its important to set quotas, rate limits, etc., on the key.

Reply:
We discussed in discord. I think Tim is right